src/java/org/apache/cassandra/io/tries/IncrementalTrieWriterPageAware.java

```diff
@@ -310,67 +310,18 @@ void completeChild(NODE child)
         }
     }
 
-    class RecalcTotalSizeRecursion extends Recursion<Node<VALUE>>
-    {
-        final long nodePosition;
-        int sz;
-
-        RecalcTotalSizeRecursion(Node<VALUE> node, Recursion<Node<VALUE>> parent, long nodePosition)
-        {
-            super(node, node.children.iterator(), parent);
-            sz = 0;
-            this.nodePosition = nodePosition;
-        }
-
-        @Override
-        Recursion<Node<VALUE>> makeChild(Node<VALUE> child)
-        {
-            if (child.hasOutOfPageInBranch)
-                return new RecalcTotalSizeRecursion(child, this, nodePosition + sz);
-            else
-                return null;
-        }
-
-        @Override
-        void complete()
-        {
-            node.branchSize = sz;
-        }
-
-        @Override
-        void completeChild(Node<VALUE> child)
-        {
-            // This will be called for nodes that were recursively processed as well as the ones that weren't.
-
-            // The sizing below will use the branch size calculated above. Since that can change on out-of-page in branch,
-            // we need to recalculate the size if either flag is set.
-            if (child.hasOutOfPageChildren || child.hasOutOfPageInBranch)
-            {
-                long childPosition = this.nodePosition + sz;
-                child.nodeSize = serializer.sizeofNode(child, childPosition + child.branchSize);
-            }
-
-            sz += child.branchSize + child.nodeSize;
-        }
-    }
-
     private int recalcTotalSizeRecursive(Node<VALUE> node, long nodePosition) throws IOException
     {
         return recalcTotalSizeRecursiveOnStack(node, nodePosition, 0);
     }
 
-    private int recalcTotalSizeRecursiveOnStack(Node<VALUE> node, long nodePosition, int depth) throws IOException
+    protected int recalcTotalSizeRecursiveOnStack(Node<VALUE> node, long nodePosition, int depth) throws IOException
```

Comment:
How about dropping the `OnStack` method and the depth too?

Reply:
ok